### PR TITLE
PLAT-122618: Fix Spinner change color based on day/night mode

### PR DESCRIPTION
**Pull request by adrian-cocoara-lgp**
Spinner changes color based on day/night mode
Fix `transparent` knob from sampler

Enact-DCO-1.0-Signed-off-by: Adrian Cocoara adrian.cocoara@lgepartner.com

### CODE REVIEW

File: Spinner/Spinner.js [hunk 1]
```diff
@@ -169,16 +167,18 @@ const SpinnerBase = kind({
 	},
 
 	computed: {
-		className: ({children, color, paused, size, styler, transparent}) => styler.append(
-			color,
+		className: ({children, skinVariants, paused, size, styler, transparent}) => styler.append(
```
Comment (daniel-stoian-lgp) [inline Spinner/Spinner.js:170]: please sort

**Review by stanca-pop-lgp — APPROVED** [review]

**Review by daniel-stoian-lgp — APPROVED** [review]

**Review by ybsung — APPROVED** [review]
Hello @adrian-cocoara-lgp 

Related with the `color` changes,
I thought we'd apply `Skinnable({variantsProp: 'skinVariants'})` in not a Spinner component but a Spinner sampler. We couldn't remove any props until releasing Agate 2.0. So it would be better to fix the Spinner sampler. I fixed it.

Anyway, thanks for your idea and code. Due to your effort, we could make our spinner sampler nice.

LGTM

File: samples/sampler/stories/default/Spinner.js [hunk 2]
```diff
@@ -1,15 +1,32 @@
...
+import PropTypes from 'prop-types';
 import {storiesOf} from '@storybook/react';
 
+import Skinnable from '@enact/agate/Skinnable';
 import Spinner, {SpinnerBase} from '@enact/agate/Spinner';
 
 Spinner.displayName = 'Spinner';
 const Config = mergeComponentMetadata('Spinner', UiSpinnerBase, UiSpinner, SpinnerBase, Spinner);
 
+const SkinSpinnerBase = kind({
```
Comment (0x64) [inline samples/sampler/stories/default/Spinner.js:16]: @adrian-cocoara-lgp
It looks better to rename this to `SkinnedSpinnerBase` just like a SkinnedIcon in Icon sampler and a SkinnedButton in a button sampler. :)

File: samples/sampler/stories/default/Spinner.js [hunk 3]
```diff
@@ -1,15 +1,32 @@
...
+	propTypes: {
+		skinVariants: PropTypes.object
+	},
+
+	render: ({skinVariants, ...rest}) => {
+		return <Spinner {...rest} color={skinVariants.night ? 'light' : 'dark'} />;
+	}
+});
+
+const SkinSpinner = Skinnable({variantsProp: 'skinVariants'}, SkinSpinnerBase);
```
Comment (0x64) [inline samples/sampler/stories/default/Spinner.js:28]: So, `SkinnedSpinner` here.

**Review by 0x64 — COMMENTED** [review]
Please check the comments below, and please resolve conflicts for merging. Looks good except for names. 👍

**Comment by adrian-cocoara-lgp** [comment]
@0x64 I made the changes you requested. Could you please take a look? Thanks!

**Pull request merged**